src/main/java/com/ucl/imaginethisserver/Controller/UserController.java

```diff
@@ -0,0 +1,43 @@
+package com.ucl.imaginethisserver.Controller;
+
+import com.ucl.imaginethisserver.Model.User;
+import com.ucl.imaginethisserver.Service.UserService;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.bind.annotation.*;
+
+import java.util.HashMap;
+import java.util.Map;
+import java.util.UUID;
+
+@RestController
+@CrossOrigin("http://localhost:3000")
+@RequestMapping("/api/v1")
+public class UserController {
+    private final UserService userService;
+
+    @Autowired
+    public UserController(UserService userService) {
+        this.userService = userService;
+    }
+
+    @PostMapping("/users")
+    public ResponseEntity<Map<String, Object>> createNewUser(@RequestBody User user) {
+        UUID userId = userService.createUser(user);
+        Map<String, Object> response = new HashMap<>();
+        response.put("success", true);
+        response.put("userId", userId);
+        return new ResponseEntity<>(response, HttpStatus.OK);
+    }
+
+    @PatchMapping("/users/{user-id}")
```

Comment:
Just one thing, I'd use `PUT` instead of `PATCH`. Although it's my personal preference, I think it should be that in this case as well, see [SO answer](https://stackoverflow.com/a/34400076/6418493), since the request updates the whole user, not just those fields that are sent.